pyfr/integrators/dual/pseudosteppers.py

```diff
@@ -13,32 +13,27 @@ def collect_stats(self, stats):
         # Total number of pseudo-steps
         stats.set('solver-time-integrator', 'npseudosteps', self.npseudosteps)
 
-    def _add_with_dts(self, *args, c):
-        vals, regs = list(args[::2]), list(args[1::2])
+    def _rhs_with_dts(self, t, uin, fout, c=1):
+        # Compute -∇·f
+        self.system.rhs(t, uin, fout)
 
         # Coefficients for the dual-time source term
         svals = [c*sc for sc in self._dual_time_source]
 
-        # Normal addition
-        axnpby = self._get_axnpby_kerns(len(vals))
-        self._prepare_reg_banks(*regs)
-        self._queue % axnpby(*vals)
-
-        # Source addition
+        # Source addition -∇·f - dQ/dt
         axnpby = self._get_axnpby_kerns(len(svals) + 1, subdims=self._subdims)
-        self._prepare_reg_banks(regs[0], self._idxcurr, *self._source_regidx)
+        self._prepare_reg_banks(fout, self._idxcurr, *self._source_regidx)
```

Comment:
It is correct, since we changed the function from add_with_dts to rhs_with_dts. In the former everything was added to the first register regs[0]. Now the function adds the source to fout.